consensus/src/chained_bft/consensus_types/timeout_certificate.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::chained_bft::common::{self, Author, Round};
+use failure::prelude::*;
+use network;
+use proto_conv::{FromProto, IntoProto};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, fmt};
+use types::{
+    account_address::AccountAddress,
+    crypto_proxies::{Signature, ValidatorVerifier},
+};
+
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq)]
+/// TimeoutCertificate is a proof that 2f+1 participants have voted in round r and we can now move
+/// to round r+1.
+pub struct TimeoutCertificate {
+    round: Round,
+    signatures: HashMap<Author, Signature>,
+}
+
+impl fmt::Display for TimeoutCertificate {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "TimeoutCertificate[round: {}]", self.round)
+    }
+}
+
+#[allow(dead_code)]
+impl TimeoutCertificate {
+    /// Creates new TimeoutCertificate
+    pub fn new(round: Round, signatures: HashMap<Author, Signature>) -> Self {
+        Self { round, signatures }
+    }
+
+    /// Verifies the signatures for the round
+    pub fn verify(&self, validator: &ValidatorVerifier) -> failure::Result<()> {
+        // TODO: should use ValidatorVerifier::batch_verify_aggregated_signature(), but Signature
+        // types don't match.
```

Comment:
@huitseeker this feels like we should have a proxy of BatchedSignature and use for both LI and TC

Reply:
This code will change after #1149 anyway.